### DISCUSSION TITLE
Incorrect response with nested Pydantic models

### DISCUSSION THREAD

**Discussion by Pandede** [fastapi/fastapi#6341] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, UploadFile
from datetime import datetime
from typing import List
from pydantic import BaseModel, Field


app = FastAPI()


class BoundingBox(BaseModel):
    x1: int
    y1: int
    x2: int
    y2: int

class ImageMeta(BaseModel):
    timestamp: datetime
    bounding_boxes: List[BoundingBox]

    class Config:
        orm_mode = True

@app.post('/fetch_image')
def fetch_image(content: UploadFile, metadata: ImageMeta):
    return {
        'filename': content.filename,
        'timestamp': metadata.timestamp.isoformat(timespec='seconds'),
        'bounding_boxes': metadata.bounding_boxes
    }
```

### Description

The Pydantic models are stated in example code, and this is the request body as shown below:
![image](https://user-images.githubusercontent.com/38829000/157482508-766d5b02-2d85-4e4e-8ef5-23b305b268dc.png)

But receive the following response, stated that the field `timestamp` is missing. For elimating this issue, I've done with following attempts:

- Reduce the field `timestamp` and only field `bounding_boxes`: got the same response stated that the field `bounding_boxes` is missing.
- Cancelled out `orm_mode`: No changes.
- Assign the default value for `timestamp` and `bounding_boxes`: the value of fields will always be default, never be modified by the incoming request.

I'm struggling with this situation for few hours and still seeking for the issue. Does anyone confront the same issue?
![image](https://user-images.githubusercontent.com/38829000/157483185-bb1d959d-e101-4637-9733-a6ae933f1c76.png)

### Operating System

Windows

### FastAPI Version

0.75.0

### Python Version

3.10.1

**Comment by Pandede** [comment]
I've found out a strange situation:
if I reduce argument `content: UploadFile`, It works fine! I'm confused.

**Comment by Pandede** [comment]
OK, I just found out `Body` and `UploadFile` cannot exist at the same time as the limitation of HTTP protocol, therefore I change the `Body` elements into `Form` data, but there is still an issue about `bounding_boxes`: It seems the field cannot be parsed as form-data.

Here is my modification:
```python
@app.post('/fetch_image')
def fetch_image(content: UploadFile = File(...), timestamp: datetime = Form(...), bounding_boxes: List[BoundingBox] = Form(...)):
    return {
        'filename': content.filename,
        'timestamp': timestamp.isoformat(timespec='seconds'),
        'bounding_boxes': bounding_boxes
    }
```

**Comment by gmanny** [comment]
Can you post an example curl request that supplies `bounding_boxes` that can't be parsed?

**Comment by Pandede** [comment]
> Can you post an example curl request that supplies `bounding_boxes` that can't be parsed?

```bash
curl -X 'POST' \
  'localhost:8000/fetch_image' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'content=@sample.png;type=image/png' \
  -F 'metadata={
  "timestamp": "2022-03-18T01:11:28.591Z",
  "bounding_boxes": [
    {
      "x1": 0,
      "y1": 0,
      "x2": 0,
      "y2": 0
    }
  ]
```

**Comment by dotX12** [comment]
https://github.com/dotX12/pyfa-converter


```python3
from datetime import datetime
from typing import Optional

from fastapi import FastAPI, UploadFile, File, Form
from pydantic import BaseModel, Field

from pyfa_converter import FormDepends, PyFaDepends

app = FastAPI()


class PostContractBodySchema(BaseModel):
    title: str = Field(..., description="Description title")
    date: Optional[datetime] = Field(
        None, description="Example: 2021-12-14T09:56:31.056Z"
    )


@app.post("/form-data-body")
async def example_foo_body_handler(
    data: PostContractBodySchema = FormDepends(PostContractBodySchema),
    # data1: PostContractBodySchema = PyFaDepends( # OR
    #         model= PostContractBodySchema, _type=Form
    #     ),
    document: UploadFile = File(...),
):
    return {"title": data.title, "date": data.date, "file_name": document.filename}

```

**Comment by dotX12** [comment]
@Pandede, Above solution :)

**Comment by Pandede** [comment]
> https://github.com/dotX12/pyfa-converter
> 
> ```python
> from datetime import datetime
> from typing import Optional
> 
> from fastapi import FastAPI, UploadFile, File, Form
> from pydantic import BaseModel, Field
> 
> from pyfa_converter import FormDepends, PyFaDepends
> 
> app = FastAPI()
> 
> 
> class PostContractBodySchema(BaseModel):
>     title: str = Field(..., description="Description title")
>     date: Optional[datetime] = Field(
>         None, description="Example: 2021-12-14T09:56:31.056Z"
>     )
> 
> 
> @app.post("/form-data-body")
> async def example_foo_body_handler(
>     data: PostContractBodySchema = FormDepends(PostContractBodySchema),
>     # data1: PostContractBodySchema = PyFaDepends( # OR
>     #         model= PostContractBodySchema, _type=Form
>     #     ),
>     document: UploadFile = File(...),
> ):
>     return {"title": data.title, "date": data.date, "file_name": document.filename}
> ```

Excellent repository, thumb up for the elegant solution
